src/dynamo/models/user.ts

```diff
@@ -0,0 +1,199 @@
+const emailAttr: keyof UnhydratedUser = 'email';
+
+const client = createClient({
+  name: 'USERS',
+  partitionKey: idAttr,
+  attributes: {
+    [idAttr]: 'S',
+    [usernameLowerAttr]: 'S',
+    [emailAttr]: 'S',
+  },
+  indexes: [
+    {
+      partitionKey: usernameLowerAttr,
+      name: 'ByUsername',
+    },
+    {
+      partitionKey: emailAttr,
+      name: 'ByEmail',
+    },
+  ],
+});
+
+const stripSensitiveData = (user: UserWithSensitiveInformation): UnhydratedUser => {
+  const sanitized = { ...user };
+  //@ts-expect-error -- Typescript says property must be optional to delete, but we are switching types so not relevant
+  delete sanitized.passwordHash;
+  //@ts-expect-error --  Ditto
+  delete sanitized.email;
+
+  return sanitized;
+};
+
+const batchStripSensitiveData = (users: UserWithSensitiveInformation[]): UnhydratedUser[] =>
+  users.map(stripSensitiveData);
+
+const hydrate = (user: UnhydratedUser): User => {
+  const hydrated = { ...user } as User;
+  hydrated.image = getImageData(hydrated.imageName || 'Ambush Viper');
+
+  if (!hydrated.defaultPrinting) {
+    hydrated.defaultPrinting = DefaultPrintingPreference;
+  }
+  if (typeof user.autoBlog === 'undefined') {
+    user.autoBlog = false;
+  }
+
+  if (!hydrated.gridTightness) {
+    hydrated.gridTightness = DefaultGridTightnessPreference;
+  }
+
+  return hydrated;
+};
+
+const batchHydrate = (users: UnhydratedUser[]): User[] => users.map(hydrate);
+
+const getByUsername = async (
+  username: string,
+  lastKey?: Record<string, NativeAttributeValue>,
+): Promise<User | null> => {
+  const result = await client.query({
+    IndexName: 'ByUsername',
+    KeyConditionExpression: `#p1 = :uname`,
+    ExpressionAttributeValues: {
+      ':uname': username.toLowerCase(),
+    },
+    ExpressionAttributeNames: {
+      '#p1': usernameLowerAttr,
+    },
+    ExclusiveStartKey: lastKey,
+  });
+
+  if (result.Items && result.Items.length > 0) {
+    return hydrate(stripSensitiveData(result.Items[0] as UserWithSensitiveInformation));
+  }
+
+  return null;
+};
+
+const user = {
+  getById: async (id: string): Promise<User | undefined> => {
+    const result = await client.get(id);
+    if (!result.Item) {
+      return undefined;
+    }
+    return hydrate(stripSensitiveData(result.Item as UserWithSensitiveInformation));
+  },
+
+  getByIdWithSensitiveData: async (id: string): Promise<UserWithSensitiveInformation | undefined> =>
+    (await client.get(id)).Item as UserWithSensitiveInformation,
+
+  getByUsername,
+
+  getByIdOrUsername: async (idOrUsername: string): Promise<User | null> => {
+    const result = await client.get(idOrUsername);
+
+    if (result.Item) {
+      return hydrate(stripSensitiveData(result.Item as UserWithSensitiveInformation));
+    }
+
+    return getByUsername(idOrUsername);
+  },
+
+  getByEmail: async (email: string, lastKey?: Record<string, NativeAttributeValue>): Promise<User | null> => {
```

Comment:
I think this needs to return UserWithSensitiveInfo, or strip the sensitive info

Reply:
You and the AI got it right. Changed to strip the sensitive issue and updated a couple of places in the forgot password flow which used getByEmail and assumed the user returned contained an email field (which is sensitive)